### PR TITLE
fix optional params

### PR DESCRIPTION
**Pull request by barkerxuan**
This PR removes dots from URI path and fixes optional parameters(`vendorId` and `version`) when get financial product config from finance core

### CODE REVIEW

File: tests/FinanceCoreApiTest.php [hunk 1]
```diff
@@ -28,79 +28,94 @@ class FinanceCoreApiTest extends \PHPUnit\Framework\TestCase
...
+        'defaultPaymentCycle' => 'weekly',
+        'invoiceRequired' => true,
+        'manualSettlementRequired' => true,
+        'version' => 1,
+    ];
+
+    private $expectedConfigResponse = [
+        'data' => [
+            'financialProductConfiguration' => [
+                'establishmentFee' => 4.99,
```
Comment (darkbluesun) [inline tests/FinanceCoreApiTest.php:54]: I forget my PHP, is it possible to use $this->expectedConfig here?

**Review by darkbluesun — DISMISSED** [review]

File: tests/FinanceCoreApiTest.php (hunk 1 above)
Comment (barkerxuan) [reply tests/FinanceCoreApiTest.php:54]: I tried but it raised an error: Constant expression contains invalid operations

Comment (darkbluesun) [reply tests/FinanceCoreApiTest.php:54]: alright. I guess it's not possible. I'm just worried about the repetition.
could you perhaps construct this expectedconfigresponse in the constructor?

**Review by darkbluesun — APPROVED** [review]
Nice work :ok_hand:

**Pull request merged**